File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        unordered_map<int,int>mp;
        vector<int> result;

        for(auto i : nums){
            mp[i]++;
        }
        for(auto i : mp){
            if(i.second>=2){
                result.push_back(i.first);
            }
        }
        return result;
    }
};

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> ans = new ArrayList<>();
        Map<Integer,Integer> mp = new HashMap<>();

        for(int num : nums){
            mp.put(num,mp.getOrDefault(num,0)+1);

            if(mp.get(num)>=2){
                ans.add(num);
            }
        }
        return ans;
    }
}